Cloud Storage: Bazelize some tests

oleiman opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

cloud_storage/tests:
- cloud_storage_e2e_test **[PASS]**
- delete_records_e2e_test **[PASS]**
- read_replica_test **[PASS]**
- segment_chunk_hydration_test **[PASS]**
- topic_recovery_service_test **[PASS]**

cloud_storage_clients/tests:
- client_pool_mt_test **[PASS]**

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

* none

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

oleiman committed:
cs/t: bazelize some tests

- cloud_storage_e2e_test
- delete_records_e2e_test
- read_replica_test
- segment_chunk_hydration_test
- topic_recovery_service_test

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman committed:
csc/t: bazelize a test

- client_pool_mt_test

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman commented:
Kind of a weird situation here. Couple of these tests are broken. I'd like to build them w/ cmake to confirm, but it seems the cmake build is also broken (for these tests anyway, `ada` errors). 😢

oleiman commented:
/ci-repeat 1

WillemKauf commented:
>but it seems the cmake build is also broken (for these tests anyway, ada errors).

That's because of this change which altered core code and the bazel build but not CMake: https://github.com/redpanda-data/redpanda/commit/a3235260f4f6a17518618fd31465e7e38ef3f19f

oleiman committed:
cs/t: Fix apparent race w/ filesystem in read_replica_e2e_fixture

Data directory would be removed before segments had a chance to shutdown,
leading to filesystem errors in 'disk_log_impl'.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman commented on src/v/cloud_storage/tests/read_replica_e2e_fixture.h:
```diff
@@ -50,7 +50,7 @@ class read_replica_e2e_fixture
           test_directory(),
           app.sched_groups,
-          true,
+          false,
```
not 100% clear that this is the right thing

rockwotj replied:
I also just did this for another test. I don't really understand why the directory is being destroyed twice

oleiman replied:
Yeah it's weird. In this case it only seems to happen for the fixture instance whose lifetime is scoped to the test case. I don't really get it, but it also doesn't really seem worth blocking on.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--aa6d4f8a-8b92-4925-b692-0506a345e1ca-->
<!--TEST RESULTS-->
<!--cff22368-fd5b-4207-8aac-9e5f72e45a55-->
<!--TEST RESULTS-->
<!--795cdff3-58f6-4250-9f86-983653e522c7-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65795">build#65795</a></summary>


| test_class               | test_method            | test_arguments                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:-------------------------|:-----------------------|:-----------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeTransactionTests | test_with_transactions | {"cloud_storage_type": 1, "compaction": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65795#0196bd9a-10ae-42ab-b4c6-08e5f1c4dc04 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65801">build#65801</a></summary>


| test_class                   | test_method                                         | test_arguments                                                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-----------------------------|:----------------------------------------------------|:------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AccessControlListTestUpgrade | test_upgrade_sasl                                   |                                                                                                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65801#0196c2f7-54f0-4799-906d-cc33b88689b9 | FLAKY         | 20/21    | upstream reliability is '99.18032786885246'. current run reliability is '95.23809523809523'. drift is 3.94223 and the allowed drift is set to 50. The test should PASS |
| CloudStorageChunkReadTest    | test_read_when_segment_size_smaller_than_chunk_size |                                                                                                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65801#0196c2f7-54ef-40f8-8faf-f9db98f5a237 | FLAKY         | 20/21    | upstream reliability is '99.10313901345292'. current run reliability is '95.23809523809523'. drift is 3.86504 and the allowed drift is set to 50. The test should PASS |
| DataMigrationsApiTest        | test_migrated_topic_data_integrity                  | {"params": {"cancellation": {"dir": "in", "stage": "executed"}, "use_alias": true}, "transfer_leadership": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65801#0196c2f7-54f0-4128-a372-3c7d20d92348 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              |
| PartitionReassignmentsTest   | test_add_partitions_with_inprogress_reassignments   |                                                                                                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65801#0196c2f7-54f1-4891-b01c-89089bccfc3e | FLAKY         | 17/21    | upstream reliability is '90.11406844106465'. current run reliability is '80.95238095238095'. drift is 9.16169 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65873">build#65873</a></summary>


| test_class                | test_method                    | test_arguments                                                                                                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:--------------------------|:-------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeTransactionTests  | test_with_transactions         | {"cloud_storage_type": 1, "compaction": false}                                                                                                                    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65873#0196c822-b2c5-4d2d-9682-e721d9bea073 | FLAKY         | 20/21    | upstream reliability is '99.53703703703704'. current run reliability is '95.23809523809523'. drift is 4.29894 and the allowed drift is set to 50. The test should PASS |
| PartitionMoveInterruption | test_cancelling_partition_move | {"compacted": false, "recovery": "no_recovery", "replication_factor": 3, "unclean_abort": false}                                                                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65873#0196c822-b2c4-487b-aad5-1bfb9fde4c94 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              |
| RandomNodeOperationsTest  | test_node_operations           | {"cloud_storage_type": 1, "enable_failures": false, "mixed_versions": false, "with_chunked_compaction": true, "with_iceberg": true, "with_tiered_storage": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65873#0196c821-d663-4c7d-a1d5-033e7bdb73e3 | FLAKY         | 19/21    | upstream reliability is '96.875'. current run reliability is '90.47619047619048'. drift is 6.39881 and the allowed drift is set to 50. The test should PASS            |
</details>
<!--/body-->

oleiman commented:
[force push](https://github.com/redpanda-data/redpanda/compare/0351a3a8c8d3ac9d919f3e4aa5a1fa5a68935d48..2b10407c09eae17641a298dd0432f655ebd0d0d7) errant TODOs

oleiman commented:
CI Failures:
- [build-release](https://buildkite.com/redpanda/redpanda/builds/65801#0196c2b9-d610-4f48-b4f8-6741dd8d830a) looks to be a real failure in `cloud_storage_e2e_test::ManualWithOverride/CloudStorageEndToEndManualTest.TestTimequeryAfterArchivalGC/0`. reproducible locally but not previously known afaict

oleiman committed:
cs/t: Force arbitrary disk stats in cloud_storage_e2e_test

Seems TestTimequeryAfterArchivalGC was racing against a disk status notification
for the cloud cache, occasionally leading to an exception in cache::trim.

Not entirely clear why this didn't flake in cmake land. This is fine as a hack
though, since the cache itself isn't under test. We're just trying to clear
it out to ensure we actually query block storage.

oleiman commented:
[force push](https://github.com/redpanda-data/redpanda/compare/2b10407c09eae17641a298dd0432f655ebd0d0d7..ae1a15154700444b9dded24a43baa608602700f9) force disk stats into the cloud cache to avoid racing against an async notification

oleiman commented:
CI Failure:
- [debug-ducktape](https://buildkite.com/redpanda/redpanda/builds/65854#0196c6e7-14ce-4da7-8615-eff0be79a980) looks like infra. no failed tests or anything.

WillemKauf commented:
>[debug-ducktape](https://buildkite.com/redpanda/redpanda/builds/65854#0196c6e7-14ce-4da7-8615-eff0be79a980) looks like infra. no failed tests or anything.

Should be fixed by https://github.com/redpanda-data/redpanda/pull/26109, try repeating the failed jobs one more time :)

oleiman commented:
> > [debug-ducktape](https://buildkite.com/redpanda/redpanda/builds/65854#0196c6e7-14ce-4da7-8615-eff0be79a980) looks like infra. no failed tests or anything.
> 
> Should be fixed by #26109, try repeating the failed jobs one more time :)

Ah, I didn't check the python logs. Thanks :D

oleiman commented:
/ci-repeat 1
skip-redpanda-build
skip-units

oleiman commented:
/ci-repeat 1

rockwotj commented on src/v/cloud_storage/tests/cloud_storage_e2e_test.cc:
```diff
@@ -702,6 +702,14 @@ TEST_P(CloudStorageEndToEndManualTest, TestTimequeryAfterArchivalGC) {
       archiver->manifest().get_spillover_map().begin()->base_offset,
       first_seg.base_offset);
 
+    // Supply some phony disk stats so that the cache doesn't panic and
+    // think it has zero bytes of space. Otherwise we seem to be racing
+    // against an async notification in release mode.
+    app.shadow_index_cache.local().notify_disk_status(
+      100ULL * 1024 * 1024 * 1024,
+      50ULL * 1024 * 1024 * 1024,
+      storage::disk_space_alert::ok);
```
I don't understand this at all, might be good to get someone with more context on this test to review :)

dotnwat replied:
could this have anything to do with the isolated test environment that bazel provides?

oleiman replied:
maybe? but it's not deterministic and doesn't reproduce in debug mode afaict. still I don't _think_ this really affects the semantics of the test. wdyt?

dotnwat replied:
yeh, seems like a race

oleiman replied:
indeed

oleiman replied:
@dotnwat you feel good to merge these or should i get a second opinion?

andrwng replied:
It's odd that this isn't being seen at all in CMake. Agree with others this just seems like a race between local node monitor (on a 1s interval) and the test performing the trim. Wondering if maybe bazel's isolation means that the stat is slower or somesuch.

In any case, this fix lgtm; we do it in a few other tests
